services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py

```diff
@@ -0,0 +1,121 @@
+from ..models.schemas.files import File
+from ..models.schemas.jobs import ArgumentTypes, Job, JobInputs, JobOutputs
+from ..models.schemas.studies import Study, StudyID
+from .storage import to_file_api_model
+
+
+class ProjectInputs(NamedTuple):
+    inputs: list[ProjectInputUpdate]
+    file_inputs: dict[InputID, InputTypes]
+
+
+def get_project_and_file_inputs_from_job_inputs(
+    project_inputs: dict[NodeID, dict[str, Any]],
+    file_inputs: dict[InputID, InputTypes],
+    job_inputs: JobInputs,
+) -> ProjectInputs:
+    job_inputs_dict = job_inputs.values
+
+    # TODO make sure all values are set at some point
+
+    for name, value in job_inputs.values.items():
+        if isinstance(value, File):
+            # FIXME: ensure this aligns with storage policy
+            file_inputs[InputID(name)] = SimCoreFileLink(
+                store=0,
+                path=SimcoreS3FileID(f"api/{value.id}/{value.filename}"),
+                label=value.filename,
+                eTag=value.e_tag,
+            )
+
+    new_inputs: list[ProjectInputUpdate] = []
+    for node_id, node_dict in project_inputs.items():
+        if node_dict["label"] in job_inputs_dict:
+            new_inputs.append(
+                ProjectInputUpdate(
+                    key=node_id, value=job_inputs_dict[node_dict["label"]]
+                )
+            )
+
+    return ProjectInputs(new_inputs, file_inputs)
+
+
+async def create_job_outputs_from_project_outputs(
+    job_id: StudyID,
+    project_outputs: dict[NodeID, dict[str, Any]],
+    user_id,
+    storage_client,
+) -> JobOutputs:
+    results: dict[str, ArgumentTypes] = {}
+
+    for _, node_dict in project_outputs.items():
+        name = node_dict["label"]
+        value = node_dict["value"]
+        if (
+            value and isinstance(value, collections.abc.Mapping) and "store" in value
+        ):  # TODO make this more robust
+            path = value["path"]
+            file_id: UUID = File.create_id(*path.split("/"))
+
+            found = await storage_client.search_files(
+                user_id=user_id,
+                file_id=file_id,
+                sha256_checksum=None,
+                access_right="read",
+            )
+            if found:
+                assert len(found) == 1  # nosec
+                results[name] = to_file_api_model(found[0])
+            else:
+                api_file: File = await storage_client.create_soft_link(
+                    user_id, path, file_id
+                )
+                results[name] = api_file
+        else:
+            results[name] = value
+    job_outputs = JobOutputs(job_id=job_id, results=results)
+    return job_outputs
+
+
+def create_job_from_study(
+    study_key: StudyID,
+    project: ProjectGet,
+    job_inputs: JobInputs,
+) -> Job:
+    """
+    Given a study, creates a job
+
+    raise ValidationError
+    """
+
+    study_name = f"{study_key}"  # TODO do we use the study name here?
```

Comment:
there is a todo left here

Reply:
I removed this code for another PR, hence resolving this conversation